[4400] Implement sidebar widget mechanics

aachavarria opened the pull request:
https://github.com/craftercms/craftercms/issues/4400

aachavarria committed:
removing siteLinks from GlobalNav.tsx;

aachavarria committed:
Refactors on ToolsPanel.tsx to read the new Widgets from state;

rart commented on ui/app/src/models/GlobalState.ts:
```diff
@@ -115,6 +114,9 @@ export interface GlobalState {
     showToolsPanel: boolean;
+    toolsPanelPageStack: {
+      pages: any[];
+    };
```
Are we thinking this may have additional props? Why not it being directly the array of "pages"

aachavarria replied:
yes, I think is better to apply it directly to toolsPanelPageStack

rart commented on ui/app/src/models/GlobalState.ts:
```diff
@@ -179,11 +181,8 @@ export interface GlobalState {
-      siteNav: {
-        links: SiteNavConfigEntry[];
+      toolsPanel: {
+        widgets: any[];
```
This is typeable. Please type.

rart replied:
👆🏻

rart commented on ui/app/src/modules/Preview/Tools/SiteExplorer.tsx:
```diff
@@ -27,11 +27,10 @@ import { useActiveSiteId, useLogicResource, useRoles } from '../../../utils/hook
-  supported: SidebarPanelConfigEntry[];
-  notSupported: SidebarPanelConfigEntry[];
+  supported: any[];
+  notSupported: any[];
```
Fix types

rart commented on ui/app/src/modules/Preview/Tools/SiteExplorer.tsx:
```diff
@@ -161,7 +160,7 @@ export function SiteExplorer(props: SiteExplorerProps) {
 
 interface SiteExplorerContainerProps {
-  widgets: SidebarPanelConfigEntry[];
+  [key: string]: any;
```
Fix types

rart commented on ui/app/src/modules/Preview/ToolsPanel.tsx:
```diff
@@ -148,112 +75,33 @@ const translations = defineMessages({
-  tools: any[];
-}
-
-function UnknownPanel(props: UnknownPanelProps) {
-  const classes = useStyles({});
-  return (
-    <ToolPanel title={translations.unknownPanelTitle}>
-      <Typography component="div" variant="body1" className={`${classes.panelBodyInner} ${classes.center}`}>
-        <div>
-          <WarningRounded />
-        </div>
-        <pre className={classes.ellipsis} title={props.id}>
-          {props.id}
-        </pre>
-        <FormattedMessage
-          id="previewTools.unknownPanel"
-          defaultMessage={`This panel is unknown to the system. Please check your configuration.`}
-        />
-      </Typography>
-    </ToolPanel>
-  );
-}
```
We still need this guy

rart commented on ui/app/src/state/actions/preview.ts:
```diff
@@ -306,3 +306,8 @@ export const showSystemNotification = createAction<{
+
+// region toolsPanelPageStack
+export const pushPageStack = createAction<any>('PUSH_PAGE_STACK');
+export const popPageStack = createAction<any>('POP_PAGE_STACK');
```
(push|pop)ToolsPanelPage

aachavarria committed:
adding pushToolsPanelPage to ToolsPanelPageButton.tsx

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/4400

 Conflicts:
	ui/app/src/components/Navigation/GlobalNav.tsx
	ui/app/src/models/GlobalState.ts
	ui/app/src/state/actions/preview.ts

aachavarria committed:
ToolsPanel.tsx: using uiConfigDefaults.ts to setup the widgets;
Doing changes on PreviewPanels;
Fixing wrong translations;

aachavarria committed:
updating uiConfigDefaults.ts;
Moving PreviewPanels to its own folder;
Deleting oldCode;
Updating BrowseComponentsPanel.tsx ReceptaclesPanel.tsx and InPageInstancesPanel.tsx

aachavarria committed:
adding icons for siteTools;
adding new prop showChildrenRail to PathNavigator.tsx;
updates on panels to support state restore;

jvega190 committed:
Create EmbeddedLegacySiteTools component, validate modules action when in embedded mode

rart committed:
Sync with rart/3.2/stability branch

rart committed:
Snackbar styles so it doesn't overlap topbar

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/4400

aachavarria committed:
adding segments to uiConfigDefaults.ts
updating PreviewAudiencesPanel.tsx to load config from props;

aachavarria committed:
reading xml;

rart commented on ui/app/src/components/Widget/Widget.tsx:
```diff
@@ -51,6 +54,8 @@ const Widget = memo(function(props: WidgetProps) {
       return <NonReactWidget widget={record} configuration={configuration} />;
     }
+  } else if (!plugin) {
+    return <EmptyState title={formatMessage(messages.componentNotFound)} styles={{ image: { width: 100 } }} />;
```
Add component id to message. Message should end up as: 
empty state title: Component "{id}" not found.
empty state subtitle: Check ui config & make sure you've installed the plugins that contain the desired components.

rart commented on ui/app/src/components/ToolsPanelListItemButton/ToolsPanelListItemButton.tsx:
```diff
@@ -43,14 +44,15 @@ export default function ToolsPanelListItemButton(props: ToolsPanelListItemButton
+        <SystemIcon icon={icon} />
       </ListItemIcon>
-      <ListItemText primary={title} secondary={subtitle} />
+      <ListItemText primary={formatMessage({ id: title, defaultMessage: title })} secondary={subtitle} />
```
Please test if the built app shows this without throwing exceptions. Need to test in default locale (en) and any other too (e.g. es) to make sure.

rart commented on ui/app/src/components/PreviewComponentsPanel/PreviewComponentsPanel.tsx:
```diff
@@ -114,22 +114,60 @@ export const ComponentsPanelUI: React.FC<ComponentsPanelUIProps> = (props) => {
+            ]
+          }
+        })
+      ])
```
Let's move the logic of using a the ToolsPanelPage to an util function that receives configuration `{ title: '', widgets: [...] }` and returns this whole chunk.

```js
pushToolsPanelPage(createToolsPanelPage({
  title: 'previewBrowseComponentsPanel.title',
  widgets: [...]
}))
```

rart commented on ui/app/src/components/Navigation/PathNavigator/styles.ts:
```diff
@@ -76,6 +76,10 @@ export const useStyles = makeStyles(
       },
+      childrenRail: {
+        marginLeft: 10,
+        borderLeft: `3px solid ${palette.gray.light2}`
```
Does the color work well on both theme modes? 
Try `theme.palette.paper.default` or divider color

rart commented on ui/app/src/components/Dialogs/CreateFileDialog.tsx:
```diff
@@ -140,7 +140,7 @@ function CreateFileUI(props: CreateFileUIProps) {
                 <FormattedMessage id="newFile.controller.required" defaultMessage="Controller name is required." />
               ) : (
-                <FormattedMessage id="newFile.controller.required" defaultMessage="Template name is required." />
+                <FormattedMessage id="newFile.template.required" defaultMessage="Template name is required." />
```
newFile.templateRequired

Change all other keys to respect the convention

rart committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into 3.2.0/stability

aachavarria committed:
PR review;
Fixing issues on PreviewComponentsPanel.tsx and PreviewInPageInstancesPanel.tsx;

rart committed:
Fix bug in initial state fetcher when there aren't any sites in the system.

rart committed:
Handle errors in plugin loading via Widget

aachavarria committed:
fixing wrong icon;

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/4400

jvega190 committed:
Rename component to LegacySiteToolsFrame, add tool as optional, rename workarea to workareaOnly

rart committed:
Improve error handling message in plugin load via Widget

rart committed:
Add styles to avoid layout issues with long messages in ErrorState in narrow containers

rart committed:
Merge branch 'feature/4400' of https://github.com/aachavarria/studio-ui into feature/4400

 Conflicts:
	ui/app/src/components/CrafterCMSNextBridge.tsx
	ui/app/src/components/Widget/Widget.tsx

rart committed:
Remove duplicate action showSystemNotification from actions/system

rart committed:
Remove console.log

rart committed:
Create/move component to it's folder

rart committed:
Cleanup

rart committed:
Cleanup

rart committed:
Cleanup

rart committed:
Merge branch 'feature/4413' of https://github.com/jvega190/studio-ui into feature/4400

rart committed:
Merge/integrate LegacySiteToolsFrame, fix snackbar/toolbar spacing so toolbar is interactive, misc tweaks

rart committed:
Avoid error in style creation

aachavarria committed:
Merge branch 'feature/4400' of https://github.com/aachavarria/studio-ui into feature/4400

rart committed:
Using consistent ids, update SystemIcon to use component registry instead of it's own map

aachavarria committed:
Merge branch 'feature/4400' of https://github.com/aachavarria/studio-ui into feature/4400

rart committed:
Bugfix: infinite loop of quick create fetch calls when no site is selected

rart committed:
Merge branch 'feature/4400' of https://github.com/aachavarria/studio-ui into feature/4400
